OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -139,20 +144,31 @@ void RenderThread(object contextObject)
 			}
 		}
 
-		internal Vertex[] GetVertices(int size)
+		internal T[] GetVertices<T>(int size)
 		{
-			lock (verticesPool)
-				if (size <= BatchSize && verticesPool.Count > 0)
-					return verticesPool.Pop();
+			lock (vertexBufferPools)
+			{
+				Stack<T[]> pool;
+				if (!vertexBufferPools.TryGetValue(typeof(T), out var poolObject))
+				{
+					pool = new Stack<T[]>();
+					vertexBufferPools.Add(typeof(T), pool);
+				}
+				else
+					pool = poolObject as Stack<T[]>;
```

Comment:
Use a direct cast: `(Stack<T[]>)poolObject`. If the cast fails we want it to throw early, not give us a null.

Reply:
I had tried that at first and it didn't work. It seems to work fine now